pixel_dancer.py

```diff
@@ -0,0 +1,693 @@
+"""
+Project : Olin College Software Design Spring 2017
+By : Seungin Lyu and Yichen Jiang
+Game Name : Pixel_Dancer
+Version : 0.1
+Date : March 8th, 2017
+
+Instruction : Move your doggy around with the arrow keys to paint as many
+pictures as possible. Remember, you should follow the music rhythm! If you
+aren't following the beat, you will lose your precious energy and end up
+only coloring half the grid you are on.  Also, you don't want to go out of the
+picture and be careful not to eat the toxic chocolates! You will gain some
+extra energy everytime you complete one picture. Good Luck!
+(by the way..the energy decreases naturally because doggy is hungry..)
+You can play with the game settings by manipulating the defined constants.
+
+Things to FIX:
+1) __init__ method of all viewer classes should only specify the self.model
+attribute and they should extend it from a basic Viewer class.
+Also, draw() methods for each classes should be defined.
+Right now the classes aren't technically classes.
+
+2) OOP(Object Oriented Programming) Features for classes such as:
+   - Gridlist has Grids.
+   - All the different Viewer classes extend default Viwer class.
+
+"""
+
+import pygame
+import numpy
+import random
+
+'''Global variable declaration'''
+
+# PREDEFINED COLORS
+WHITE = (255, 255, 255)
+BLACK = (0, 0, 0)
+BLUE = (0, 0, 255)
+RED = (255, 0, 0)
+YELLOW = (255, 255, 0)
+
+# GAME SETTINGS
+NUM_BG = 8  # Number of Background Pictures
+NUM_X = 4  # Number of Cols
+NUM_Y = 4  # Number of Rows
+NUM_MONSTER = 5  # Number of Monsters
+TOTAL_GRID = NUM_X * NUM_Y  # Total Number of Grids
+FPS = 60  # frames(while loops) per second
+MAX_ENERGY = 100  # Maximum Energy for the Doggy
+ENERGY_CONSTANT = 120  # energy lasts ENERGY_CONSTANT seconds(maximum)
+DAMAGE = int(0.1 * MAX_ENERGY)
+BONUS = int(0.2 * MAX_ENERGY)
+MARGINAL_ERROR = 10  # the marginal error for following the beat
+
+# MUSIC PRESETS
+MUSIC = 'Ghost_fight.mp3'  # the background music
+BEAT_CONST = 64  # arbitrary Frames/BEAT constant
+
+
+class Grid(object):
+    '''
+    This class represents a single grid in the game canvas.
+    '''
+    def __init__(self, row, col, location, length, width, image,
+                 alpha=256):
+        """
+        This method initializes the Grid Object
+
+        Precondition : 1) Relative Row, Col number in a gridlist,
```

Comment:
Little note on nomenclature: precondition is a condition of a program that has to be satisfied before this function is executed for it to run properly. In this context, 'parameters' is more appropriate term than precondition.